### PR TITLE
DL-2568 BTA Banner

### PR DESCRIPTION
**Pull request by Sir-Iyke**
DL-2568 Added BTA Banner on all pages

**New feature**

Created ServiceInfoPartialConnector which goes through appConfig to get the BTA Banner. ServiceInfoPartialConnector then passes this to ServiceInfoService. ServiceInfoService checks if the user is an agent. If they are then the BTA banner doesn't get shown. I then injected ServiceInfoService into every controller so that I could pass it into all the "view" methods. The "view" methods will display the page with the retrieved BTA banner.

## Checklist

*Iyke* 
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [x]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

*Reviewer* (Replace with your name)
 - [ ]  I've confirmed that every effort has been made to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've confirmed appropriate tests has been included with any code added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [ ]  I've confirmed code was added using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

**Comment by Sir-Iyke** [comment]
Could not put BTA banner on the unauthorised page because ServiceInfoService requires implicit StandardAuthRetrievals. StandardAuthRetrievals is used to check if the user is an agent or not. The BTA banner only displays to user's who are not agents.

image

**Comment by Sir-Iyke** [comment]
Also could not put BTA banner on the feedback page, possibly due to the same reason as above

File: app/services/MetricsService.scala [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/*
```
Comment (jmountain-hmrc) [inline app/services/MetricsService.scala:1]: Is this something we need as  part of this ticket?

File: app/services/EnrolmentsAuthService.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/*
```
Comment (jmountain-hmrc) [inline app/services/EnrolmentsAuthService.scala:1]: I don't think this file is needed as part of this ticket

**Comment by Sir-Iyke** [comment]
@jmountain-hmrc MetricsService and EnrolmentAuthService are both used as part of DIModule. I've removed MetricsService and EnrolmentAuthService then when I run the unit tests I get a null pointer exception for ReliefDecleration. I'm not sure why this is the only test that fails but i'm still looking into it.

**Comment by jmountain-hmrc** [comment]
> @jmountain-hmrc MetricsService and EnrolmentAuthService are both used as part of DIModule. I've removed MetricsService and EnrolmentAuthService then when I run the unit tests I get a null pointer exception for ReliefDecleration. I'm not sure why this is the only test that fails but i'm still looking into it.

I'd probably remove them from the DIModule

**Comment by Sir-Iyke** [comment]
> > @jmountain-hmrc MetricsService and EnrolmentAuthService are both used as part of DIModule. I've removed MetricsService and EnrolmentAuthService then when I run the unit tests I get a null pointer exception for ReliefDecleration. I'm not sure why this is the only test that fails but i'm still looking into it.
> 
> I'd probably remove them from the DIModule

That's what I did which caused that one test to fail

**Pull request merged**